File: 2938/main.go
package main

import "fmt"

func main() {
	fmt.Println(minimumSteps("101"))
}

func minimumSteps(s string) int64 {
	ans := int64(0)
	sum := 0
	for i := 0; i < len(s); i++ {
		if s[i] == '1' {
			sum++
		} else {
			ans += int64(sum)
		}
	}
	return ans
}

File: 2938/main.rs
struct Solution{}

impl Solution {
    pub fn minimum_steps(s: String) -> i64 {
        let mut ans: i64 = 0;
        let mut sum = 0;
        for i in s.chars() {
            if i == '1' {
                sum += 1;
            } else {
                ans += sum as i64;
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::minimum_steps("101".to_string()))
}